src/components/Shared/GlobalIntegrationGrid/index.js

```diff
@@ -0,0 +1,178 @@
+import { Box } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { GeneralLedgerRepository } from 'src/repositories/GeneralLedgerRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useForm } from 'src/hooks/form'
+import { useResourceQuery } from 'src/hooks/resource'
+
+const GlobalIntegrationGrid = ({ masterSource, masterId}) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const {
+    labels: _labels,
+    access,
+  } = useResourceQuery({
+    datasetId: ResourceIds.IntegrationAccount,
+  })
+
+  const { formik } = useForm({
+    maxAccess: access,
+    enableReinitialize: false,
+    validateOnChange: true,
+    initialValues: {
+      Integrations: [
+        {
+          id: 1,
+          postTypeId: null,
+          ptName: '',
+          accountId: null,
+          accountRef: null,
+          accountName: ''
+        }
+      ]
+    },
+    onSubmit: values => {
+      postIntegrations(values);
+    }
+  });
+  
+  const postIntegrations = values => {
+    
+    const filteredIntegrations = values.Integrations.filter(integration => integration.accountId !== null);
+
+    const data = { 
+      masterSource: masterSource,
+      masterId: masterId,
+      integrationAccounts: filteredIntegrations
+    };
+
+    postRequest({
+      extension: GeneralLedgerRepository.IntegrationAccounts.set2,
+      record: JSON.stringify(data)
+    })
+    .then(res => {
+      toast.success('Record Successfully Saved');
+    })
+    .catch(error => { });
+  };
+  
+    const column = [
+      {
+        component: 'textfield',
+        label: _labels.ptName,
+        name: 'ptName',
+        props:{readOnly: true}
+      },
+      {
+        component: 'resourcecombobox',
```

Comment:
it should be a lookup not a combobox
sometimes there is over 1000 accounts. it's heavy to load them all in one step. lookups are better in this case
the lookup fills from GL.asmx/snapshotAC